gateway-ha/src/main/java/io/trino/gateway/ha/persistence/JdbcConnectionManager.java

```diff
@@ -60,13 +64,29 @@ public Jdbi getJdbi(@Nullable String routingGroupDatabase)
                 .registerRowMapper(new RecordAndAnnotatedConstructorMapper());
     }
 
-    private String buildJdbcUrl(@Nullable String routingGroupDatabase)
+    @VisibleForTesting
+    String buildJdbcUrl(@Nullable String routingGroupDatabase)
     {
         String jdbcUrl = configuration.getJdbcUrl();
-        if (routingGroupDatabase != null) {
-            jdbcUrl = jdbcUrl.substring(0, jdbcUrl.lastIndexOf('/') + 1) + routingGroupDatabase;
+        if (routingGroupDatabase == null) {
+            return jdbcUrl;
+        }
+        try {
+            int index = jdbcUrl.indexOf("/") + 1;
+            URI uri = new URI(jdbcUrl.substring(index));
+            URI newUri = new URI(
+                    uri.getScheme(),
+                    uri.getUserInfo(),
+                    uri.getHost(),
+                    uri.getPort(),
+                    Path.of(uri.getPath()).resolveSibling(routingGroupDatabase).toString(),
+                    uri.getQuery(),
+                    uri.getFragment());
+            return jdbcUrl.substring(0, index) + newUri;
+        }
+        catch (URISyntaxException e) {
+            throw new RuntimeException(e);
```

Comment:
Hey @ebyhr WDYT of catching more broder exception like RuntimeException or Exception
Because NPE could happen if jdbcUrl is none, or IndexOutOfBounds if / is not present, or IllegalArgumentException if `Path.of(uri.getPath())` is malformed path. 

```suggestion 
    catch (Exception e) { // catch all
        Logger.get(getClass()).error(e, "Failed to build JDBC URL for routing group '%s'", routingGroupDatabase);
        throw new RuntimeException(e);
    }
```

Reply:
Do you think it's better to do catching more brother exception?
The exceptions you mentioned are treated as if statements, so I think it looks good now